services/services.go

```diff
@@ -637,27 +637,80 @@ func getIndexPageData() (*types.IndexPageData, error) {
 
 	// get epochs from db
 	var epochs []*types.IndexPageDataEpochs
-	err = db.ReaderDb.Select(&epochs, `SELECT epoch, finalized, globalparticipationrate, votedether FROM epochs ORDER BY epochs DESC LIMIT 10`)
+	err = db.ReaderDb.Select(&epochs, `
+		SELECT 
+			epoch, 
+			finalized, 
+			globalparticipationrate, 
+			votedether,
+
+			eligibleether,
+			validatorscount,
+			averagevalidatorbalance
+		FROM epochs 
+		ORDER BY epochs DESC 
+		LIMIT 10`)
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving index epoch data: %v", err)
 	}
 	minEpoch := epochs[len(epochs)-1].Epoch
+
+	if len(epochs) > 0 {
+		for i := len(epochs) - 1; i >= 0; i-- {
+			if epochs[i].Finalized {
+				data.CurrentFinalizedEpoch = epochs[i].Epoch
+				data.FinalityDelay = FinalizationDelay()
+				data.AverageBalance = string(utils.FormatBalance(uint64(epochs[i].AverageValidatorBalance), currency))
+				break
+			}
+		}
+
+		data.StakedEther = string(utils.FormatBalance(epochs[len(epochs)-1].EligibleEther, currency))
+		data.ActiveValidators = epochs[len(epochs)-1].ValidatorsCount
```

Comment:
This code now is duplicated (see code below from lines 684 to 694). I suggest putting this in a not-exported function that takes the epoch slice (I guess you would have to use **IndexPageDataEpochs** for both cases and get rid of **IndexPageEpochHistory**) and the **IndexPageData** as parameters.

Reply:
Implemented in new commit